platforms/documentation/docs/src/snippets/dependencyManagement/artifactTransforms-simple/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,39 @@
+import org.gradle.api.artifacts.transform.TransformParameters
+
+plugins {
+    application
+}
+
+// tag::artifact-transform-imp[]
+abstract class MyTransform : TransformAction<TransformParameters.None> {
+    @get:InputArtifact
+    abstract val inputArtifact: Provider<FileSystemLocation>
+
+    override fun transform(outputs: TransformOutputs) {
+        val inputFile = inputArtifact.get().asFile
+        val outputFile = outputs.file(inputFile.name.replace(".jar", "-transformed.jar"))
+        // Perform transformation logic here
+    }
+}
+// end::artifact-transform-imp[]
+
+// tag::artifact-transform-use[]
+val artifactType = Attribute.of("artifactType", String::class.java)
```

Comment:
We should use `ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE` (which is the same thing as this)